src/components/video/platforms/YoutubePlayer.vue

```diff
@@ -0,0 +1,124 @@
+<script setup lang="ts">
+import { computed } from 'vue';
+import { Icon } from '@iconify/vue/dist/iconify.js';
+import { useI18n } from 'vue-i18n';
+
+import messages from '@/i18n/video';
+import Duration from '../Duration.vue';
+
+const props = defineProps<{
+  url: string;
+  startTime?: number;
+  endTime?: number;
+}>();
+
+const i18n = useI18n({
+  messages,
+  useScope: 'local',
+});
+const { t } = i18n;
+
+const isUrlValid = computed(() => {
+  try {
+    videoIdFromURL(props.url);
+    return true;
+  } catch {
+    return false;
+  }
+});
+
+function embedURLfromRegularURL(url_: string, startTime_?: number, endTime_?: number): string {
+  if (!isUrlValid.value) {
+    return '';
+  }
+
+  const prefix: string = 'https://www.youtube-nocookie.com/embed/';
+
+  const videoID: string = videoIdFromURL(url_);
+
+  let suffix: string = startTime_ || endTime_ ? '?version=3&loop=1&modestbranding=1&' : '';
+  suffix += startTime_ ? 'start=' + startTime_ : '';
+  suffix += startTime_ && endTime_ ? '&' : '';
+  suffix += endTime_ ? 'end=' + endTime_ : '';
+
+  console.log(prefix + videoID + suffix);
+
+  return prefix + videoID + suffix;
```

Comment:
use `URL()` instead

Also… why are the params suffixed with `_`?

Reply:
Fixed in new commits of @WaldemarLehner.